delegation-toolkit/get-started/eip7715-quickstart.md

```diff
@@ -0,0 +1,162 @@
+```
+
+### 4. Request ERC-7715 permissions
+
+Request ERC-7715 permissions from the user. Currently, only the 
+`native-token-stream` permission type is supported, which allows the dApp to stream 
+native tokens from the user's wallet.
+
+```typescript
+const expiry = Math.floor(Date.now() / 1000 + 604_800); // 1 week from now.
+const currentTime = Math.floor(Date.now() / 1000); // now
+
+const grantedPermissions = await walletClient.grantPermissions([{
+  chainId: chain.id,
+  expiry,
+  signer: {
+    type: "account",
+    data: {
+      address: sessionAccount.address,
+    },
+  },
+  permission: {
+    type: "native-token-stream",
+    data: {
+      initialAmount: 1n, // 1 wei
+      amountPerSecond: 1n, // 1 wei per second
+      maxAmount: 10n, // 10 wei
+      startTime: currentTime,
+      justification: "Payment for a week long subscription",
+    },
+  },
+}]);
+```
+
+### 5. Set up a Bundler Client
+
+Set up a [Viem Bundler Client](https://viem.sh/account-abstraction/clients/bundler) 
+using Viem's `createBundlerClient` function. This lets you use the bundler service 
+to estimate gas for user operations and submit transactions to the network.
+
+Once you’ve created the client, make sure to extend the Bundler Client 
+functionality using `erc7710BundlerActions`. These actions enable you to redeem ERC-7715 permissions, and execute transactions on a user's behalf. 
+
+```typescript
+import { createBundlerClient } from "viem/account-abstraction";
+import { erc7710BundlerActions } from "@metamask/delegation-toolkit/experimental";
+
+const bundlerClient = createBundlerClient({
+  client: publicClient,
+  transport: http("https://your-bundler-rpc.com"),
+  // Allows you to use the same Bundler Client as paymaster.
+  paymaster: true
+}).extend(erc7710BundlerActions());
+```
+
+### 6. Redeem ERC-7715 permissions
+
+The session account can now [redeem the delegation](../how-to/redeem-delegation.md). The redeem transaction is sent to the `DelegationManager` contract, which validates the delegation and executes actions on the user's behalf.
+
+To redeem the permissions, you can use the `sendUserOperationWithDelegation` bundler client action.
+
+```typescript
+// These properties must be extracted from the permission response.
+const permissionsContext = grantedPermissions[0].context;
+const delegationManager = grantedPermissions[0].signerMeta.delegationManager;
+const accountMetadata = grantedPermissions[0].accountMeta;
+
+// Calls without permissionsContext and delegationManager will be executed 
+// as a normal user operation.
+const userOperationHash = await bundlerClient.sendUserOperationWithDelegation({
+  publicClient,
+  account: sessionAccount,
+  calls: [
+    {
+      to: sessionAccount.address,
+      data: "0x",
+      value: 1n,
+      permissionsContext,
+      delegationManager,
+    },
+  ],
+  // Appropriate values must be used for fee-per-gas. 
+  maxFeePerGas: 1n,
+  maxPriorityFeePerGas: 1n
+  accountMetadata,
+});
+```
```

Comment:
I see that this page has some overlap with [ERC-7715: Request Permissions](https://docs.metamask.io/delegation-toolkit/experimental/erc-7715-request-permissions/) and [ERC-7710: Redeem delegations](https://docs.metamask.io/delegation-toolkit/experimental/erc-7710-redeem-delegations/). Would it make sense to consolidate this content or refer to those experimental guides somewhere on this page?

Reply:
Yeah, but we need to improve ERC-7715, since with the addition of new permissions, each will have its own dedicated section. This way, the page content won’t be overlapping.